app/routes/login.tsx

```diff
@@ -65,7 +65,7 @@ export const action: ActionFunction = async ({ request }) => {
     request,
     userId: user.id,
     remember: remember === "on" ? true : false,
-    redirectTo: typeof redirectTo === "string" ? redirectTo : "/notes",
+    redirectTo,
```

Comment:
Maybe inline the call, so we're first doing redirects before "calculating" it?

```suggestion
    redirectTo: safeRedirect(formData.get("redirectTo"), "/notes"),
```